site/content/integrate/apps/structure/forms/_index.md

```diff
@@ -0,0 +1,712 @@
+
+Example `lookup` call response:
+
+```json
+{
+    "type": "ok",
+    "data": {
+        "items": [
+            {
+                "label": "Option One",
+                "value": "option_1"
+            },
+            {
+                "label": "Option Two",
+                "value": "option_2"
+            }
+        ]
+    }
+}
+```
+
+##### Select field refresh
+
+If the `refresh` value is set to `true`, the form's `source` call is performed any time the field's value changes. The call request will include the App, user, channel, and team IDs in the context; the current values of the form are also included.
+The call response is expected to contain a full, updated form definition.
+
+#### Markdown fields
+
+Markdown fields are a special field that allows you to better format your form. They will not generate any value in the form submission sent to the App. The content is defined in the `description` property of the field.
+
+### Form submission
+
+When the form is submitted, either by executing a slash command or clicking a submit button, the form's `submit` call is performed.
+The call request will include the App, user, channel, and team IDs in the context; the current values of the form are also included.
+
+## Slash command arguments
+
+Slash command arguments and flags are defined by form fields. When a slash command is typed, the command arguments are retrieved from the command's form.
+If a form was not included with the command binding, the binding's call will be invoked to provide a form response.
+The call request will include the App, user, post, root post (if any), channel, and team IDs in the context.
+
+{{<collapse id="slash-command-flags-example" title="Example form with flag arguments">}}
+```json
+{
+    "location": "sub",
+    "label": "sub",
+    "description": "Subscribe to an event",
+    "form": {
+        "title": "Subscribe to an event",
+        "header": "Subscribe to a Mattermost Server event",
+        "icon": "icon.png",
+        "fields": [
+            {
+                "name": "eventname",
+                "label": "eventname",
+                "type": "text",
+                "subtype": "input",
+                "description": "The name of the event to subscribe to",
+                "is_required": true
+            },
+            {
+                "name": "teamid",
+                "label": "teamid",
+                "type": "text",
+                "subtype": "input",
+                "description": "The ID of the team"
+            },
+            {
+                "name": "channelid",
+                "label": "channelid",
+                "type": "text",
+                "subtype": "input",
+                "description": "The ID of the channel"
+            }
+        ],
+        "submit": {
+            "path": "/sub",
+        }
```

Comment:
If a `call` has just a `path` with no other fields, it can be shortened to just a string:

```suggestion
        "submit": "/sub"
```

Reply:
Also, trailing/dangling commas are not allowed in JSON data, as opposed to data defined in Go/JS source code

Reply:
IMO This should have been documented in the godoc for the `Call` struct's `UnmarshalJSON` function. I'm not convinced one would think there's a custom unmarshaller that can accept either a struct or a simple string by seeing `*Call` as the type.

It might have made more sense to have the type of a call be `any` or `interface{}` so there's no ambiguity; the reader knows the type is `interface{}` and would need to find the code that determines its underlying type.

Reply:
What do you think about using the `string` form in our example docs when pnly a `path` is being provided?

Reply:
My first concern is the confusion that Go devs may encounter: Is it a string, or isn't it?
From a technical perspective, it works just fine the way it is. From a cognitive perspective, however, the load is greater.

My suggestion is to leave the example as-is.